lib/src/Uncertainty/Distribution/JointDistribution.cxx

```diff
@@ -344,17 +387,17 @@ JointDistribution * JointDistribution::clone() const
 Point JointDistribution::getRealization() const
 {
   const UnsignedInteger dimension = getDimension();
+  // Complex case: use 1D CDF inversion
+  if (!kernel_.isCopula() || !hasIndependentCopula())
+  {
+    Point realization(kernel_.getRealization());
+    for (UnsignedInteger i = 0; i < dimension; ++i) realization[i] = distributionCollection_[i].computeScalarQuantile(realization[i]);
+    return realization;
+  } // General case
   if (dimension == 1) return distributionCollection_[0].getRealization();
   // Special case for independent copula
   Point result(dimension);
-  if (hasIndependentCopula())
-  {
-    for (UnsignedInteger i = 0; i < dimension; ++i) result[i] = distributionCollection_[i].getRealization()[0];
-    return result;
-  }
-  // General case
-  const Point realization(copula_.getRealization());
-  for (UnsignedInteger i = 0; i < dimension; ++i) result[i] = distributionCollection_[i].computeScalarQuantile(realization[i]);
+  for (UnsignedInteger i = 0; i < dimension; ++i) result[i] = distributionCollection_[i].getRealization()[0];
```

Comment:
I am surprised by this bit of code. Maybe I missed it, but I don't see where you handle the case where `kernel_`*is* a copula. I saw:

- the complex case where the `kernel_`is not a copula *and* the copula is not independent
- the trivial 1d case
- the independent copula case

Where do you handle the cases where:
- `kernel_`is not a copula but has an independent copula
- `kernel_`is a copula but is not independent